packages/nimble-components/src/wafer-map/specs/README.md

```diff
@@ -0,0 +1,213 @@
+This means that we might be displaying several `nimble-wafer-map` components on a single page each containing several hundred or thousands of data points, each having the individual functionality mentioned in the "Features" section of this spec document.
+We are planning to perform initial benchmarking on this aspect, but not planning to optimize it in the first release.
+
+### Prior Art/Examples
+
+Please see below a few screenshots regarding the current `wafer-map` component:
+
+![WaferMap Example](./Resources/wafer_map_example.png)
+
+## Design
+
+From a user’s perspective one of the most important actions which this component should be able to achieve is to zoom in and out on the datapoints. This functionality should occur when the mouse pointer hovers the `nimble-wafer-map` component and a mouse wheel action is performed.
+While zooming in on the component, the user should also be able to pan (drag) the wafer map by holding the left mouse button and moving the cursor on the XY axis.
+
+From a developer perspective, it's very important that the `nimble-wafer-map` component can be integrated with the MicroStrategy Custom plugin framework as this framework provides functionalities as follows:
+* Tooltip when hovering any if the datapoints withing the wafer map
+* Wafer map legend
+* MicroStrategy specific context menus on right-click action
+
+Information regarding the basic MicrosStrategy **Custom Plugin Workflow** can be found [here](https://www2.microstrategy.com/producthelp/Current/VisSDK/Content/topics/HTML5/Understanding_the_workflow.htm)
+Also, the visualisation widget has no active controls, so there is no way to alter the visual aspect or data post rendering. Every input parameter must be provided through the constructor of the wafer map component.
+
+### API
+
+*The key elements of the component's public API surface:*
+* Component Name: `nimble-wafer-map`
+* Props/Attrs: 
+  * `die[]` - this represents the input data, which fills the wafer map with content.\
+  The **die**  object contains the following attributes:
+	* x: integer
+	* y: integer
+	* alue: float
+  * `quadrant` - represents the orientation of the dies on the wafer map (the layout of the values on the dies). It can be represented by an Enum with the following values: 
+	* TopLeft - ![Top Left Quadrant](./Resources/top_left.png)
+	* BottomLeft - ![Bottom Left Quadrant](./Resources/bottom_left.png)
+	* TopRight - ![Top Right Quadrant](./Resources/top_right.png)
+	* BottomRight - ![Bottom Right Quadrant](./Resources/bottom_right.png)
+  * `orientation` - represent the orientation of the notch on the wafer map outline (only visual). As only four static orientations are possible, it can be represented by an Enum with the following values: top, bottom, left, right.
+  * `colorScale` - represents the color spectrum which shows the status of the dies on the wafer.\
```

Comment:
Continuing the discussion from [this thread](https://github.com/ni/nimble/pull/805#discussion_r1017187303) in a new one so GitHub associates it with the new location in the file...

Thanks for clarifying the plan. I think I'd advocate not to expose d3's types as part of the public API and rather see if we can find a standard way to represent the data that can be converted to d3's types. That way we're not tied to d3 if we decide to change our underlying implementation.

It looks like the scale ordinal is just a pair of arrays for color and values. Could we just make those be two separate properties?

Reply:
Conceptually I don't like them as two different properties, because they are fundamentally associated with one another. One idea (and I'm just shooting from the hip) is coming up with an interface (e.g., `IColorScale`) that would return a color (likely hex string) for a given numeric value. We could then implement that interface with a custom element, which could then be slotted into the wafer-map component:

```
<wafer-map>
    <d3-color-scale slot="color-scale" colors="red, blue, green" values="1, 2, 3">
</wafer-map>
```

Reply:
I spoke with the dev team on a call this morning. Having a separate slotted object to represent a scale seems like an interesting long term direction and we should record it in the spec as an alternative. It would be especially useful if we want to include more information in a scale. 

But for the first pass to reduce scope growth we think two arrays is an ok place to start. We can still switch to the scale object in the future, it would just be a breaking change (not a big deal since we'll know about all the clients initially).